File: jdroid-java-firebase-database/src/main/java/com/jdroid/java/firebase/database/auth/CustomTokenFirebaseAuthenticationStrategy.java
package com.jdroid.java.firebase.database.auth;

import com.firebase.client.Firebase;

public abstract class CustomTokenFirebaseAuthenticationStrategy extends FirebaseAuthenticationStrategy {

	@Override
	protected void doAuthenticate(Firebase firebase) {
		firebase.authWithCustomToken(getAuthToken(), this);
	}

	protected abstract String getAuthToken();
}

File: jdroid-java-firebase-database/src/main/java/com/dipien/java/firebase/database/auth/CustomTokenFirebaseAuthenticationStrategy.kt
package com.dipien.java.firebase.database.auth

import com.firebase.client.Firebase

abstract class CustomTokenFirebaseAuthenticationStrategy : FirebaseAuthenticationStrategy() {

    protected abstract val authToken: String

    override fun doAuthenticate(firebase: Firebase) {
        firebase.authWithCustomToken(authToken, this)
    }
}
